### DISCUSSION TITLE
EF Core Migrations: .NET CLI Commands

### DISCUSSION THREAD

**Discussion by bricelam** [bricelam/bricelam.github.io#87] [General]
# EF Core Migrations: .NET CLI Commands

One of the new platforms that we’re targeting in Entity Framework Core is ASP.NET Core. With this new platform comes a new set of challenges for how we enable the Entity Framework commands. Ever since Entity Framework 4.3, we’ve provided a set of PowerShell commands that could be run in Visual Studio from NuGet’s Package Manager Console. However, that won’t help you if you’re developing on OSX where neither PowerShell nor Visual Studio are available. This post will show you how to use the new Entity Framework .NET Core CLI Commands.

https://www.bricelam.net/2014/09/14/migrations-on-k/

**Comment by bricelam** [comment]
_From **RobertK** on October 8, 2014 17:58_

<p>Could you help me out on this one. I did everything described in the post above, but when I enter <br>"k ef help migration" in the cmd I'm getting the following message: <br>"'EntityFramework.Commands' does not contain a static 'Main' method suitable for an entry point".<br>The following package was downloaded from myget: "7.0.0-beta1-11140".</p><p>I searched the internet over and over again, but was unable to find a proper solution. Just like I said earlier, I hope you could help me out!</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on October 23, 2014 21:02_

<p>Did you ever find a solution, Robert? Things are evolving quickly, and it may just work now.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **RobertK** on October 29, 2014 19:56_

<p>Sorry for my late reply.<br>No, I have updated my project.json to use the latest version of EntityFramework.Commands, but I am still receiving the same error. The version that was downloaded is "7.0.0-beta2-11484".</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on October 29, 2014 21:31_

<p>Try upgrading your KRE.<br></p><pre>SET KRE_NUGET_API_URL=https://www.myget.org/F/aspnetvnext/api/v2<br>kvm upgrade</pre>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Matt** on January 7, 2015 23:48_

<p>Hello again, I can't get the EF commands to work still, when I run "kpm restore" it finds my .json file, writes a bunch of lines that start with "CACHE " and then the following line:<br>-----<br>Resolving complete, 448ms elapsed<br>kpm : 'npm' is not recognized as an internal or external command,<br>At line:1 char:1<br>+ kpm restore<br>+ ~~~~~~~~~~~<br>    + CategoryInfo          : NotSpecified: ('npm' is not re...ternal command,:String) [], RemoteException<br>    + FullyQualifiedErrorId : NativeCommandError<br>operable program or batch file.<br>-----</p><p>I am trying to get to the point of being able to add a migration in the ASP.NET vNext starter web template. Any help would be greatly appreciated and your posts have been very helpful!</p><p>Also I have already ran "kvm upgrade" and that was successful and tried the following command to add the path as recommended from here (<a href="http://forums.asp.net/t/2026904.aspx?Running+KPM+RESTORE+can+t+find+NPM+or+GRUNT):" rel="nofollow noopener" title="http://forums.asp.net/t/2026904.aspx?Running+KPM+RESTORE+can+t+find+NPM+or+GRUNT):">http://forums.asp.net/t/202...</a></p><p>$env:Path = $env:Path + ";C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Web Tools\External"</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **Matt** on January 8, 2015 0:32_

<p>Manually installed nodejs to get npm, had to add a 'npm' folder to the roaming directory to resolve an error, and installed grunt through npm. The errors continue, but I have gotten past the above issues!</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[James Chambers](https://disqus.com/by/CanadianJames)** on May 4, 2015 12:24_

<p>Just a note for anyone reading this since the release of Visual Studio 2015 (or earlier if you have a build released since they renamed Project K).</p><p>I believe the new commands would be as follows (replacing the ones from above), still to be run from the project root:</p><p>    dnu restore<br>    cd src\YourProject<br>    dnx . ef [options] [command]</p><p>Hope this helps.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Pete Burgess](https://disqus.com/by/peteburgess)** on May 14, 2015 11:12_

<p>If anyone is missing the incantation to get dnu, etc on the path, the easiest option is open a VS Command Prompt and then:</p><p></p><pre><code><br>dnvm use default<br></code></pre><p></p><p>after which the dnu/dnx commands should work.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **Matt** on May 14, 2015 23:04_

<p>4 months later and back to the same thread, thanks to both of you for providing the updates here</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on May 16, 2015 3:08_

<p>Yes, thank you both. I finally got around to updating the post.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **Jay** on June 16, 2015 9:03_

<p>This wordked Pete, Thanks</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Ben](https://disqus.com/by/disqus_NN0VYyLMez)** on June 26, 2015 13:34_

<p>How do you specify which DbContext class to target when you have more than one?</p><p>EDIT: Nevermind. I'm not used to commandline stuff, discovered a simple but helpful help command that taught me how to do so</p><p>dnx . ef migration add FirstMigration -c UserContext</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Ben](https://disqus.com/by/disqus_NN0VYyLMez)** on June 26, 2015 13:43_

<p>I have an issue where I have created my EF context and classes and have then manually created a matching Table for a Property on my context. I got this working and compiling. However, when trying to add a migration and then apply it I get an error that there is already an object name 'MyTable' in the database. I cannot see any options in the command line tool to allow me to overwrite this DB. Is this not going to be possible, or is it perhaps something which is coming up?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on June 26, 2015 17:43_

<p>Can you just manually delete the database to start over? If so, something like <a href="https://github.com/aspnet/EntityFramework/issues/2476" rel="nofollow noopener" title="https://github.com/aspnet/EntityFramework/issues/2476">#2476</a> might make it easier.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[afzal](https://disqus.com/by/afzal_grt)** on August 9, 2015 18:15_

<p>tried everything microsoft is going back to 80s they're pulling every UI feature VS 2015 is also full of crap dead slow</p><p>dnx . ef migration add --help works fine</p><p>PM&gt; dnx . ef migration add MyMigration</p><p>but everything out of the box is not compatible wtf</p><p>dnx : System.InvalidOperationException: The current runtime target framework is not compatible with 'WebApplication1'.</p><p>At line:1 char:1</p><p>+ dnx . ef migration add MyMigration</p><p>+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</p><p>    + CategoryInfo          : NotSpecified: (System.InvalidO...bApplication1'.:String) [], RemoteException</p><p>    + FullyQualifiedErrorId : NativeCommandError</p><p>Current runtime Target Framework: 'DNXCore,Version=v5.0 (dnxcore50)'</p><p>  Type: CoreCLR</p><p>  Architecture: x86</p><p>  Version: 1.0.0-beta5-12103</p><p>Please make sure the runtime matches a framework specified in project.json</p><p>   at Microsoft.Framework.Runtime.DefaultHost.GetEntryPoint(String applicationName)</p><p>   at Microsoft.Framework.ApplicationHost.Program.ExecuteMain(DefaultHost host, String applicationName, String[] args)</p><p>   at Microsoft.Framework.ApplicationHost.Program.Main(String[] args)</p><p>--- End of stack trace from previous location where exception was thrown ---</p><p>   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()</p><p>   at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider</p><p>serviceProvider)</p><p>   at dnx.host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env)</p><p>   at dnx.host.RuntimeBootstrapper.ExecuteAsync(String[] args)</p><p>   at dnx.host.RuntimeBootstrapper.Execute(String[] args)</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Shimmy](https://disqus.com/by/weitzhandler)** on August 20, 2015 3:58_

<p>Is it a normal cmd console? How do I launch the console on Windows?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on August 20, 2015 16:12_

<p>Push the Windows key, type "Command Prompt", and hit Enter.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Shimmy](https://disqus.com/by/weitzhandler)** on August 21, 2015 4:23_

<p>I'm maybe not as smart as I think I'm but I'm not as stupid as you make me...<br>hehe just kidn', anyway, <i>dnx</i> and <i>dnu</i> are unrecognized.<br>When I open a new Command Prompt console and enter either of the above commands, the following output is printed: </p><pre>'dnx' is not recognized as an internal or external command,<br>operable program or batch file.</pre><p> And BTW, I'm used to launch the console by hitting the Windows key + R, typing <i>cmd</i> and hitting Enter...</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on August 22, 2015 18:13_

<p>Does <code>dnvm list</code> show anything? For more info on how to use ASP.NET 5 and DNX, see <a href="http://docs.asp.net/" rel="nofollow noopener" title="http://docs.asp.net/">docs.asp.net</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Shimmy](https://disqus.com/by/weitzhandler)** on August 22, 2015 20:40_

<p>It does. It shows some installed versions under <i>beta5</i>, and one of <i>beta6</i>, which has got the <i>default</i> alias. <a href="http://pastebin.com/nWjtWAxU" rel="nofollow noopener" title="http://pastebin.com/nWjtWAxU">Here</a>'s the complete output. I'm having <a href="http://stackoverflow.com/q/32131768/75500" rel="nofollow noopener" title="http://stackoverflow.com/q/32131768/75500">this</a> issue, I'm sure you can easily help me out with.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Balakrishna](https://disqus.com/by/disqus_2GFJExCrlX)** on September 7, 2015 4:59_

<p>just running 'dnvm upgrade' should register the PATH for dnx and dnu. @Shimmy you already might have figured it out but just in case for someone else.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Houstonian](https://disqus.com/by/disqus_iu2U3bAYVJ)** on January 21, 2016 21:58_

<p>That worked!</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Martin From](https://disqus.com/by/martinfrom)** on September 27, 2015 16:50_

<p>I initially created a Web Application project.</p><p>Tried moving my data layer to a separate project, but when I try to add migrations I keep getting the following error (below).</p><p>It seems like a StartupProduction or Startup class is needed in order to run migrations.<br>It somewhat makes sense since the ConfigureServices() have settings for the db connection.<br>But usually this would still be something that is configured in the web project and not in the *.Data project.</p><p>How do I go about this?<br>Is this intended behavior?</p><p>Is there any way to make it use the Startup class in my *.Web project when running migrations in the *.Data project?</p><p>sing context 'DatabaseContext'.</p><p>dnx : System.InvalidOperationException: A type named 'StartupProduction' or 'Startup' could not be found in assembly 'TestProject.Data'.</p><p>At line:1 char:1</p><p>+ dnx ef migrations add test</p><p>+ ~~~~~~~~~~~~~~~~~~~~~~~~~~</p><p>    + CategoryInfo          : NotSpecified: (System.InvalidO...estProject.Data'.:String) [], RemoteException</p><p>    + FullyQualifiedErrorId : NativeCommandError</p><p>   at Microsoft.AspNet.Hosting.Startup.StartupLoader.FindStartupType(String startupAssemblyName, IList`1 diagnosticMessages)</p><p>   at Microsoft.AspNet.Hosting.Internal.HostingEngine.EnsureStartup()</p><p>   at Microsoft.AspNet.Hosting.Internal.HostingEngine.EnsureApplicationServices()</p><p>   at Microsoft.AspNet.Hosting.Internal.HostingEngine.get_ApplicationServices()</p><p>   at Microsoft.Data.Entity.Commands.ContextTool.TryCreateContextFromStartup(Type type, String startupAssemblyName)</p><p>   at Microsoft.Data.Entity.Commands.ContextTool.CreateContext(Type type, String startupAssemblyName)</p><p>   at Microsoft.Data.Entity.Commands.MigrationTool.CreateContext(Type type, String startupAssemblyName)</p><p>   at Microsoft.Data.Entity.Commands.MigrationTool.AddMigration(String migrationName, String contextTypeName, String startupAssemblyName, String rootNamespace, String projectDir)</p><p>   at Microsoft.Data.Entity.Commands.Program.&lt;&gt;c__DisplayClass11_0.&lt;addmigration&gt;b__0()</p><p>   at Microsoft.Data.Entity.Commands.Program.Execute(String startupProject, Action invoke)</p><p>   at Microsoft.Data.Entity.Commands.Program.AddMigration(String name, String context, String startupProject)</p><p>   at Microsoft.Data.Entity.Commands.Program.&lt;&gt;c__DisplayClass9_7.&lt;main&gt;b__17()</p><p>   at Microsoft.Dnx.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args)</p><p>   at Microsoft.Data.Entity.Commands.Program.Main(String[] args)</p><p>--- End of stack trace from previous location where exception was thrown ---</p><p>   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()</p><p>   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)</p><p>   at Microsoft.Dnx.ApplicationHost.Program.ExecuteMain(DefaultHost host, String applicationName, String[] args)</p><p>   at Microsoft.Dnx.ApplicationHost.Program.Main(String[] args)</p><p>--- End of stack trace from previous location where exception was thrown ---</p><p>   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()</p><p>   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)</p><p>   at Microsoft.Dnx.Host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, FrameworkName targetFramework)</p><p>   at Microsoft.Dnx.Host.RuntimeBootstrapper.ExecuteAsync(String[] args, FrameworkName targetFramework)</p><p>   at Microsoft.Dnx.Host.RuntimeBootstrapper.Execute(String[] args, FrameworkName targetFramework)</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Martin From](https://disqus.com/by/martinfrom)** on September 27, 2015 23:55_

<p>I tried out the --appbase and the --lib options but none of them even seem to work</p><p>I need to somehow specifiy that the project.json file is in one project where the migrations should also be located, and the startup is in another.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[themightylemon](https://disqus.com/by/themightylemon)** on February 3, 2016 16:24_

<p>You might not care now but this stumped me all day.  Eventually I found the answer which is that you simply 'AddItem' on the data project to add a startup class.  Then use the configure services method as usual.</p><p><a href="http://www.jerriepelser.com/blog/moving-entity-framework-7-models-to-external-project" rel="nofollow noopener" title="http://www.jerriepelser.com/blog/moving-entity-framework-7-models-to-external-project">http://www.jerriepelser.com...</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Сергей Полищук](https://disqus.com/by/disqus_Bsn48Uc3pH)** on October 16, 2015 9:41_

<p>Run: dnx ef migrations add InitialCreate<br>Why does an error: "'EntityFramework.Commands' does not contain a static 'Main' method suitable for an entry point"?<br>Using EF 7.0.0-beta8</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Spartacus](https://disqus.com/by/disqus_YF2YBT6LOh)** on November 6, 2015 14:50_

<p>Hi Guys,</p><p>Can anyone help me out. I recently created a test class library using ASPNET 5 beta8 to use as data access layer. After installing all the dependencies related to EF7 when I execute <code>dnx ef</code></p><p>I get the following error:</p><p>dnx : Error: Unable to load application or execute command 'EntityFramework.Commands'. Available commands: ef.<br>At line:1 char:1</p><p>+ dnx ef</p><p>+ ~~~~~~</p><p>    + CategoryInfo          : NotSpecified: (Error: Unable t...e commands: ef.:String) [], RemoteException</p><p>    + FullyQualifiedErrorId : NativeCommandError</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[colotiline](https://disqus.com/by/colotiline)** on December 8, 2015 16:58_

<p>Hi.</p><p>You need to add "EntityFramework.Commands": "7.0.0-rc1-final"  to your dependencies.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Jeremy Conterio](https://disqus.com/by/jeremyconterio)** on November 10, 2015 18:21_

<p>When i run dnx ef migrations add mymigration i get Could not load type 'Microsoft.Framework.Configuration.Helper.ConfigurationHelper' from assembly</p><p>But i have the package installed and included in my project.json, it doens't make sense i tell you, it doesn't make sense!</p><p>this whole dnvm and dnx stuff in asp.net5 is so buggy, give me something stable please</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Steve G](https://disqus.com/by/stevegentile)** on November 19, 2015 12:49_

<p>"There may be a few rough edges, but please let us know what you think about this and other Entity Framework 7 features."<br>So, it's been, how many years since EF started, and it still has the most basic of things.  When is Postgres support coming ?  All other dev tools support multiple databases at this point.  This is version '7' and out of box it doesn't support a pg setup with jsonb ???</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[David Kiv](https://disqus.com/by/david_kiv)** on January 22, 2016 16:15_

<p>Any idea how to apply just the latest migration - Ive tried running 'dnx ef database update myMigrationName' but it says no such migration exists. Is spelt right and Ive tried all sorts of variations...</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[David](https://disqus.com/by/disqus_YrytJj7pmz)** on January 29, 2016 1:11_

<p>If you're good with the most up-to-date migration, then you don't need to specify a migration name. You can just run 'dnx ef database update'. However, when specifying the migration name, be sure to use the full name, including the datetimestamp prefix.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **Adam** on February 25, 2016 2:03_

<p>Documentation for this is all over the shop. Once you've queued your migration, you need to use "dnx ef database update"</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[David Kiv](https://disqus.com/by/david_kiv)** on March 2, 2016 15:23_

<p>Ya, tried that. throws an error as soon as it runs the first migration - object named 'x' already exists type exception. Interestingly it does still seem to create the database fully in spite of that exception being thrown. But I don't like to to see that error getting thrown up like that all the same! EFMigrationsHistory table is also empty. I would assume there would be something in there!<br>Dunno, documentation is a mess (or non-existant) as you say.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on March 3, 2016 17:17_

<p>It sounds like <code>dbContext.Database.EnsureCreated()</code> is getting called before Migrations can do its thing.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[David Kiv](https://disqus.com/by/david_kiv)** on March 9, 2016 16:46_

<p>Ah okay,</p><p>yes I have that called in my dbcontext class.<br>I didn't realise this was an issue. I've since read up a bit more on it, see this article: <a href="http://thedatafarm.com/data-access/ef7-ensurecreated-vs-migrate-methods/" rel="nofollow noopener" title="http://thedatafarm.com/data-access/ef7-ensurecreated-vs-migrate-methods/">http://thedatafarm.com/data...</a></p><p>Essentially I need to remove this EnsureCreated call. Thanks for the feedback, will be nice to get this working.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Pete Burgess](https://disqus.com/by/peteburgess)** on January 31, 2016 20:35_

<p>I can only assume that it must be because your dnx path has a space in it!</p><p>You can manually add the full folder name to the path manually, either in the command prompt (each time you open it), or in the System Properties.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Manel loubna Omani](https://disqus.com/by/manelloubnaomani)** on January 31, 2016 21:11_

<p>thank you very mutch for your promp reply, actually it was because I was using developer cmd not the default.</p><p>now please: the actual command doesn't work for me :" dnx ef migrations add InitialDatabase" neither " dnx . ef migrations add InitialDatabase" like shown in the previous comments, I have this error: "System.ArgumentNullException: Value cannot be null.<br>Parameter name: connectionString"</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Pete Burgess](https://disqus.com/by/peteburgess)** on January 31, 2016 23:09_

<p>This indicates that dnx can't read the default database connection string from the config.json file (or the JSON is incorrectly formatted or the fields names are wrong). HAve a look at <a href="https://stackoverflow.com/questions/34324948/asp-net-entity-framework-7-config-json-connectionstring" rel="nofollow noopener" title="https://stackoverflow.com/questions/34324948/asp-net-entity-framework-7-config-json-connectionstring">https://stackoverflow.com/q...</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Pete Burgess](https://disqus.com/by/peteburgess)** on September 17, 2016 20:10_

<p>I'm sorry I didn't see your reply, did you sort out your problem? </p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Brian](https://disqus.com/by/bcbeatty)** on February 19, 2016 19:10_

<p>How do you downgrade/unapply a database update?</p><p>The migration '20160219190215_table4' has already been applied to the database. Unapply it and try again. If the migration has been applied to other databases, consider reverting its changes using a new migration.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on February 19, 2016 21:55_

<p>dnx ef database update &lt;the-migration-before-the-one-you-want-to-revert&gt;</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Rafi Asraf](https://disqus.com/by/rafi_asraf)** on March 8, 2016 22:48_

<p>Trying to start a new project for the model as a class library (package), added the nuget package for EF "EntityFramework.Core": "7.0.0-rc1-final",<br>"EntityFramework.MicrosoftSqlServer": "7.0.0-rc1-final"</p><p>but when I try to add the EntityFramework.Commands I get an error:</p><p>The dependency EntityFramework.Commands 7.0.0-rc1-final does not support framework .NETPlatform,Version=v5.4<br>New to EF7 /core, any idea how to get by this error?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on March 9, 2016 2:21_

<p>Issue <a href="https://github.com/aspnet/EntityFramework/issues/4102" rel="nofollow noopener" title="https://github.com/aspnet/EntityFramework/issues/4102">#4102</a>. There's no workaround for the shipped package.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Rafi Asraf](https://disqus.com/by/rafi_asraf)** on March 9, 2016 5:54_

<p>I can see the item is closed, so is it closed but not fixed?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on March 9, 2016 6:17_

<p>It is fixed in the nightly builds, but you'll need VS 2015 Update 2 (currently in RC)</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Rafi Asraf](https://disqus.com/by/rafi_asraf)** on March 9, 2016 18:55_

<p>I have just upgraded to VS 2015 Update 2, yet still have the same error, which nightly build should I update to? the one for EF core?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Rafi Asraf](https://disqus.com/by/rafi_asraf)** on March 11, 2016 8:08_

<p>Hi Brice,<br>Can you please help me out here, I am new to EF 7, and the new .NET 5<br>I have updated VS 2015 with update 2 RC, still suffering from the same error, what did you mean when you said the nightly build? the nightly build of what?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on March 11, 2016 17:04_

<p>Oh sorry, the nightly builds of EF. You can get them from <a href="http://myget.org/gallery/aspnetcidev" rel="nofollow noopener" title="http://myget.org/gallery/aspnetcidev">http://myget.org/gallery/as...</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Jonas** on March 24, 2016 7:01_

<p>Hi!</p><p>Any ideas on how to apply ef7 migrations in an automatic deploy process? Using Octopus for instance, EF6 had migrate.exe to handle this. But what's the best approach with EF7? Do you need to install the dnx command on the machine that is going to apply the migration, or is there any other way?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on March 24, 2016 19:15_

<p>The team in charge of the VS publishing dialog is working on something using <code>dnx ef migrations script --idempotent</code> during package. We also have issue <a href="https://github.com/aspnet/EntityFramework/issues/646" rel="nofollow noopener" title="https://github.com/aspnet/EntityFramework/issues/646">#646</a> that could work. You could also write your own standalone utility; the APIs are public. Check out <a href="http://overengineer.net/generating-new-entity-framework-migrations-when-dotnet-ef-is-broken" rel="nofollow noopener" title="http://overengineer.net/generating-new-entity-framework-migrations-when-dotnet-ef-is-broken">this blog post</a> for how to get started.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Manuel Perez](https://disqus.com/by/disqus_eQHqLI9avq)** on April 4, 2016 18:47_

<p>Hello, I'm getting an System.IO.FileLoadException using 'dnx ef dbcontext scaffold &lt;connectionstring&gt; EntityFramework.MicrosoftSqlServer' what can I do? thanks</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Jared](https://disqus.com/by/disqus_cMMI7xMl9s)** on July 1, 2016 13:06_

<p>Is there a way to limit scaffolding to just one schema?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on July 1, 2016 15:55_

<p>Use the <code>--schema</code> option.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Alvin](https://disqus.com/by/disqus_UUv3jdF9zJ)** on September 5, 2016 18:51_

<p>Cool!  Just need to add a little from this post <a href="https://github.com/shawnwildermuth/ASPNETCoreRC1ToRC2/blob/a0156c8882259442df2d2b114dd5b43145044b86/src/RC1toRC2/Models/ApplicationDbContext.cs" rel="nofollow noopener" title="https://github.com/shawnwildermuth/ASPNETCoreRC1ToRC2/blob/a0156c8882259442df2d2b114dd5b43145044b86/src/RC1toRC2/Models/ApplicationDbContext.cs">https://github.com/shawnwil...</a> .   Like a treasure hunt. THANKS!</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by EstellaDepperschmidtzD** [comment]
Now, regardless of your development platform, you can easily manage EF Core migrations through the command line interface. This is especially great news for those of you developing on OSX, where PowerShell and Visual Studio are not available.
